src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,75 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            int[] data = readFile(fromFileName);
+            int totalSupply = 0;
+            int totalBuy = 0;
+            for (int amount : data) {
+                if (amount > 0) {
+                    totalSupply += amount;
+                } else {
+                    totalBuy -= amount;
+                }
+            }
```

Comment:
it looks like create report responsibility

Reply:
This part of the code only calculating the total supply and total buy amounts and the logic for the report generation remains at createReport method.

Reply:
agree with previous mentor. You reader should only READ data, not processing it. Let's move the logic related to data parsing to report-related method.